Makefile

```diff
@@ -27,32 +27,126 @@ escape_double_quote = $(subst $\",$\\",$(1))
 # Utility function to require a variable to be defined
 require_defined = $(if $(value $1),$(value $1),$(error "$1 needs to be defined"))
 
+# What implementation to build
+VARIANT ?= trivial
 # Where to put build files
 OUT ?= out
 # Where to put typescript files
 TS_OUT ?= typescript
 
 .PHONY: all clean test typescript
 
-all: \
-	$(OUT)/baseDAO.tz
+trivialDAO: VARIANT = trivial
+trivialDAO: baseDAO
+
+registryDAO: VARIANT = registry
+registryDAO: baseDAO
+
+treasuryDAO: VARIANT = treasury
+treasuryDAO: baseDAO
+
+testDAO: VARIANT = test
+testDAO: baseDAO
+
+all: $(OUT)/trivialDAO.tz $(OUT)/registryDAO.tz $(OUT)/treasuryDAO.tz $(OUT)/testDAO.tz
```

Comment:
by default `make` runs the top target that doesn't beging with a `.`

I recommend moving `all` again to the top because of this

Reply:
Done